docs/blueprints/001-installation-of-extensions.md

```diff
@@ -0,0 +1,114 @@
+## Installation of ML extensions
+
+## Overview
+
+Epic: [Installer: add official support to install extensions](https://github.com/fuseml/fuseml/issues/161)
+
+Our installer needs to be updated so that it provides a better UX regarding the installation of 3rd party tools. Currently, it installs MLFlow by default, but it has no coverage for KFServing at all, which makes for an awkward installation experience for those who want to follow our (currently solely available) tutorial on end-to-end serving with MLFlow and KFServing.
+
+The installer should make a clear distinction between basic mandatory components and extensions for 3rd party tools, but at the same time continue to provide the "all-in-one" experience of installing everything in one shot. This is a summary of needed installer changes:
+
+* installer command line arguments that the user can supply to indicate which optional extensions need to be installed (e.g. fuseml-install --extension mlflow --extension kfserving)
+* make it possible to install/uninstall extensions in follow-up installer invocations, separately from the main installation
+* MLFlow no longer installed by default
+* add support to install KFServing and KNative.
+
+## Current state
+
+`fuseml-installer` has two ways of installing components
+
+* `fuseml-installer install` - installs all necessary components (but actually _also_ MLFlow, which should be an extension)
+* `make` targets. Specific targets defined in the `Makefile` describe the installation of other components, like KFServing
+
+## Proposed solution
+
+### Makefile
+
+Get rid of specific installation targets that install something for production usage (e.g. not used by testing/development).
+
+### CLI
+
+1. Implement new option for `install` command
+
+  `fuseml-installer install [--extension <name>] [--extension-repository <path>]`
+
+  This command installs all necessary components AND extensions selected by `--extension` option. Optionally uses path specified
+  by `--extension-repository` option as a source for extensions that should be installed. Path could be local directory on URL.
+
+
+2. Implement new command for managing extensions. This could be useful for adding an extension to already configured system.
+
+  `fuseml-installer extension`
+
+  This should have several sub commands, like
+
+  * `fuseml-installer extension list` - list installed extensions
+  * `fuseml-installer extension available [--extension-repository <path>]` = list extensions available for installation
+  * `fuseml-installer extension add <name> [--extension-repository <path>]` - install selected extension
+  * `fuseml-installer extension remove <name> [--extension-repository <path>]` - ubinstalls selected extension
+
```

Comment:
Suggestion from @alfsuse :

3. Add information about extensions into the config file